### ISSUE TITLE
How can test with Store

### ISSUE THREAD

**Issue by user-fal** [dropbox/Store#282]
Hi,

I am struggle to test something with Store. May be I miss something.

Having  an repo which uses some interface for database, service...and want to test that when someone submitIdentity, datastorage.updateStatus() is called with right parameter.

![Capture d’écran 2021-07-28 à 20 38 53](https://user-images.githubusercontent.com/3034836/127377961-8507be75-c041-4433-adc3-0d55f6fd087c.png)

My test works whatever params i put in updateStatus:

![Capture d’écran 2021-07-28 à 20 41 02](https://user-images.githubusercontent.com/3034836/127378174-41fa879b-2b23-4510-aa65-90f7aff946ec.png)

Am I miss something with the library in order to test it well?

**Comment by digitalbuddha** [comment]
It looks like you are using some sort of testing/mocking library. It's unclear what the behavior of it is. Could you provide a sample/reproducible test case that fails? Alternatively, check out the tests we have in the Store or app module which should give you a better idea of how to write tests without much mocking

**Comment by user-fal** [comment]
Hi,
Thanks for quick reply. Just using the popular Mockk library for Kotlin. Will check carefully the way you test in app module.

**Comment by user-fal** [comment]
Please, just to be sure, is it the right place for all tests you have? https://github.com/dropbox/Store/tree/main/store/src/test/java/com/dropbox/android/external/store4

**Comment by digitalbuddha** [comment]
Yup that's one of the test folders. Other modules have them too. This is an example test https://github.com/dropbox/Store/blob/main/store/src/test/java/com/dropbox/android/external/store4/impl/StreamWithoutSourceOfTruthTest.kt

**Comment by user-fal** [comment]
Thanks. If I understand well, I have to test code without interact with production code (means no mocking) but using Fake Implementation like FakeFetcher and simulate the things write in my production code?

**Comment by digitalbuddha** [comment]
That would make testing easier. It would also allow you to share non working parts for me to run and verify/give feedback 😊

**Comment by user-fal** [comment]
Ok, the fact that I should write code twice is not very cheerful. But if it is the only way, I will investigate on it.  Courage to me, it promises to be sporty :).
